cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -318,28 +324,61 @@ private Future<KafkaRebalance> updateStatus(Reconciliation reconciliation,
         return Future.succeededFuture(kafkaRebalance);
     }
 
-    private RebalanceOptions.RebalanceOptionsBuilder convertRebalanceSpecToRebalanceOptions(KafkaRebalanceSpec kafkaRebalanceSpec) {
-
-        RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder = new RebalanceOptions.RebalanceOptionsBuilder();
+    /* test */ AbstractRebalanceOptions.AbstractRebalanceOptionsBuilder<?, ?> convertRebalanceSpecToRebalanceOptions(KafkaRebalanceSpec kafkaRebalanceSpec) {
+
+        AbstractRebalanceOptions.AbstractRebalanceOptionsBuilder<?, ?> rebalanceOptionsBuilder;
+        // backward compatibility, no mode specified means "full-rebalance"
+        KafkaRebalanceMode mode = Optional.ofNullable(kafkaRebalanceSpec)
+                .map(kr -> kr.getMode())
+                .orElse(KafkaRebalanceMode.FULL_REBALANCE);
+        List<Integer> brokers = Optional.ofNullable(kafkaRebalanceSpec)
+                .map(kr -> kr.getBrokers())
+                .orElse(null);
+
+        switch (mode) {
+            case ADD_BROKER:
+                rebalanceOptionsBuilder = new AddBrokerOptions.AddBrokerOptionsBuilder();
+                if (brokers != null && !brokers.isEmpty()) {
+                    ((AddBrokerOptions.AddBrokerOptionsBuilder) rebalanceOptionsBuilder).withBrokers(brokers);
+                } else {
+                    throw new IllegalArgumentException("The brokers list is mandatory when using the add-broker rebalancing mode");
```

Comment:
Side note: One advantage of making the enum name string a field of the enum class is that you can easily parameterise error messages like this (`"when using the {} rebalancing mode", ADD_BROKER.name()`) so that if you ever have to change the name (unlikely I know) you don't have to go looking for all the place you wrote it manually.